src/include/miopen/batched_transpose_sol.hpp

```diff
@@ -66,6 +66,64 @@ struct BatchedTransposeSolution
     BatchedTransposeParam kernel_param_heuristic;
 };
 
+struct TransposeSolutionDefault2Nhwc : public BatchedTransposeSolution
+{
+    TransposeSolutionDefault2Nhwc(const ExecutionContext& ctx_,
+                                  miopenDataType_t data_type_,
+                                  uint32_t n_,
+                                  uint32_t c_,
+                                  uint32_t h_,
+                                  uint32_t w_)
+        : BatchedTransposeSolution(ctx_, data_type_, n_, c_, h_ * w_)
+    {
+        assert(size_t(h_ * w_) == (size_t(h_) * size_t(w_)));
```

Comment:
I guess not all the asserts, some of them may not be required at all.
That's exactly what I wanted to discuss.

Reply:
Do we have an `MIOPEN_ASSERT` that works in Release build as well? If not, I'd like to introduce it for these situations.

Reply:
Also, I don't agree with this opposition to `asserts`. They're free. I'm open to converting some of them to `MIOPEN_THROW` but removing them doesn't help us.

Reply:
```CPP
if (size_t(h_ * w_) != (size_t(h_) * size_t(w_)))
    MIOPEN_THROW(...)
```
Actually asserts are not free, not safe and almost useless due to Release\Debug builds.

Reply:
I don't see how `asserts` are not free. Please help me understand. They are dead code in release builds and the compiler should be able to optimize them out completely.

Reply:
@JehandadKhan @amberhassaan I think that these checks must be put in all the Solvers that leverage `TransposeSolutionNhwc2Default` or `TransposeSolutionDefault2Nhwc`. Specifically, theirs `IsApplicable()` must return `false` when the necessary conditions are not met.

Maybe it is better to implement something like `IsApplicable` in `TransposeSolution*` and use it in the Solvers. Because we do not want the implementation details of `TransposeSolution*` to sneak into other parts of code.

Also it may worth making `IsApplicable` also in `BatchedTransposeSolution` (if it has its own implementation limits) and reusing it in `TransposeSolution*`.

WRT the asserts here I do not care.

Reply:
fixed in latest code.